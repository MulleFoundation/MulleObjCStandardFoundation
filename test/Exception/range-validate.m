//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifdef __MULLE_OBJC__
# import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif

//#import "MulleStandaloneObjCFoundation.h"


static void   test( NSRange range, NSUInteger length)
{
   printf( "%ld:%ld : %ld = ", (long) range.location, (long) range.length, (long) length);
   @try
   {
      MulleObjCValidateRangeAgainstLength( range, length);
      printf( "OK\n");
   }
   @catch( NSException *exception)
   {
      printf( "FAIL\n");
   }
}


int   main( int argc, char *argv[])
{
   long  i, j, k;

   for( k = -2 ; k <= 2; k++)
      for( j = -2 ; j <= 2; j++)
         for( i = -2 ; i <= 2; i++)
            test( NSMakeRange( i, j), k);

   return( 0);
}
