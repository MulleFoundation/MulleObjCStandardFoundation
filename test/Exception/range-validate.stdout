-2:-2 : -2 = FAIL
-1:-2 : -2 = FAIL
0:-2 : -2 = OK
1:-2 : -2 = FAIL
2:-2 : -2 = FAIL
-2:-1 : -2 = FAIL
-1:-1 : -2 = FAIL
0:-1 : -2 = FAIL
1:-1 : -2 = FAIL
2:-1 : -2 = FAIL
-2:0 : -2 = OK
-1:0 : -2 = FAIL
0:0 : -2 = OK
1:0 : -2 = OK
2:0 : -2 = OK
-2:1 : -2 = FAIL
-1:1 : -2 = FAIL
0:1 : -2 = OK
1:1 : -2 = OK
2:1 : -2 = OK
-2:2 : -2 = FAIL
-1:2 : -2 = FAIL
0:2 : -2 = OK
1:2 : -2 = OK
2:2 : -2 = OK
-2:-2 : -1 = FAIL
-1:-2 : -1 = FAIL
0:-2 : -1 = OK
1:-2 : -1 = OK
2:-2 : -1 = FAIL
-2:-1 : -1 = FAIL
-1:-1 : -1 = FAIL
0:-1 : -1 = OK
1:-1 : -1 = FAIL
2:-1 : -1 = FAIL
-2:0 : -1 = OK
-1:0 : -1 = OK
0:0 : -1 = OK
1:0 : -1 = OK
2:0 : -1 = OK
-2:1 : -1 = OK
-1:1 : -1 = FAIL
0:1 : -1 = OK
1:1 : -1 = OK
2:1 : -1 = OK
-2:2 : -1 = FAIL
-1:2 : -1 = FAIL
0:2 : -1 = OK
1:2 : -1 = OK
2:2 : -1 = OK
-2:-2 : 0 = FAIL
-1:-2 : 0 = FAIL
0:-2 : 0 = FAIL
1:-2 : 0 = FAIL
2:-2 : 0 = FAIL
-2:-1 : 0 = FAIL
-1:-1 : 0 = FAIL
0:-1 : 0 = FAIL
1:-1 : 0 = FAIL
2:-1 : 0 = FAIL
-2:0 : 0 = FAIL
-1:0 : 0 = FAIL
0:0 : 0 = OK
1:0 : 0 = FAIL
2:0 : 0 = FAIL
-2:1 : 0 = FAIL
-1:1 : 0 = FAIL
0:1 : 0 = FAIL
1:1 : 0 = FAIL
2:1 : 0 = FAIL
-2:2 : 0 = FAIL
-1:2 : 0 = FAIL
0:2 : 0 = FAIL
1:2 : 0 = FAIL
2:2 : 0 = FAIL
-2:-2 : 1 = FAIL
-1:-2 : 1 = FAIL
0:-2 : 1 = FAIL
1:-2 : 1 = FAIL
2:-2 : 1 = FAIL
-2:-1 : 1 = FAIL
-1:-1 : 1 = FAIL
0:-1 : 1 = FAIL
1:-1 : 1 = FAIL
2:-1 : 1 = FAIL
-2:0 : 1 = FAIL
-1:0 : 1 = FAIL
0:0 : 1 = OK
1:0 : 1 = OK
2:0 : 1 = FAIL
-2:1 : 1 = FAIL
-1:1 : 1 = FAIL
0:1 : 1 = OK
1:1 : 1 = FAIL
2:1 : 1 = FAIL
-2:2 : 1 = FAIL
-1:2 : 1 = FAIL
0:2 : 1 = FAIL
1:2 : 1 = FAIL
2:2 : 1 = FAIL
-2:-2 : 2 = FAIL
-1:-2 : 2 = FAIL
0:-2 : 2 = FAIL
1:-2 : 2 = FAIL
2:-2 : 2 = FAIL
-2:-1 : 2 = FAIL
-1:-1 : 2 = FAIL
0:-1 : 2 = FAIL
1:-1 : 2 = FAIL
2:-1 : 2 = FAIL
-2:0 : 2 = FAIL
-1:0 : 2 = FAIL
0:0 : 2 = OK
1:0 : 2 = OK
2:0 : 2 = OK
-2:1 : 2 = FAIL
-1:1 : 2 = FAIL
0:1 : 2 = OK
1:1 : 2 = OK
2:1 : 2 = FAIL
-2:2 : 2 = FAIL
-1:2 : 2 = FAIL
0:2 : 2 = OK
1:2 : 2 = FAIL
2:2 : 2 = FAIL
